src/Core/Domain/Product/SpecificPrice/Exception/CannotSetSpecificPricePrioritiesException.php

```diff
@@ -29,6 +29,8 @@
 namespace PrestaShop\PrestaShop\Core\Domain\Product\SpecificPrice\Exception;
 
 /**
+ * @todo: can we get rid of this class at all? Its not usefull as we don't have a domain case were setting priorities would fail.
```

Comment:
?

Reply:
I hope I can remove it, as Its not used anywhere. And even if it was used it was only in experimental product page for feature which was not yet released.

Reply:
Actually it could be used in `setPrioritiesForProduct` and all other related methods instead of throwing a generic `CoreException` Similarly to what we do when a product's update fail and we throw a `CannotUpdateProductException` with a dedicated error code

Actually the previous usage of this exception in the updater service seemed correct

Reply:
as mentioned in comment, I believe in this case, the only reason something fails would be some database connection/sql stmt issues, but not domain related stuff. (domain related stuff is already handled by SpecificPriceConstraintException which is thrown when "duplicate" priorities are provided).
So technically it seems that CoreException should be closer to our current convention (as we do throw it on similar occasions in any other domain). 
In addition, those generic exceptions seemed more suitable for repository. 
BUT, probably if I remove those methods from repository and put them straight into Updater service, then it should be ok to throw that dedicated exception. Maybe this way everything sums up just right :thinking:

Reply:
Moved methods from repository to updater.
Though I used to think that we want to keep sql in repositories (if that would be the case then the updater service is actually the useless one, as there are no additional logic just simple sql queries, so that would fit in repositories :smile: )